modules/core/site.js

```diff
@@ -369,6 +369,8 @@ function Hm_Modal(options) {
     };
 
     this.addFooterBtn = (label, classes, callback) => {
+        this.modalFooter.find('button').filter((_, btn) => btn.textContent.trim() === label).remove();
```

Comment:
If it is called multiple times on the same modal, there might be other issues with duplicate event handlers or other duplicated content (even if they are not present now, they can show up in the future). I think it is better if we destroy the modal properly before showing it up again (or re-initializing it again). The code that calls this seems to initialize another instance of Hm_Modal but maybe it is reusing dom components which will lead to further issues. When we create a new modal, we need to make sure it is really new in terms of DOM components.